enterprise_subsidy/apps/api/v2/views/transaction.py

```diff
@@ -0,0 +1,202 @@
+"""
+Views for the enterprise-subsidy service relating to the Transaction model
+"""
+import logging
+
+from django.utils.functional import cached_property
+from django_filters import rest_framework as drf_filters
+from drf_spectacular.utils import extend_schema, extend_schema_view
+from edx_rbac.decorators import permission_required
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from openedx_ledger.models import LedgerLockAttemptFailed, Transaction
+from rest_framework import generics, permissions, status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.exceptions import APIException, NotFound, PermissionDenied, Throttled
+
+from enterprise_subsidy.apps.api.filters import TransactionAdminFilterSet
+from enterprise_subsidy.apps.api.utils import get_subsidy_customer_uuid_from_view
+from enterprise_subsidy.apps.api.v1.serializers import (
+    TransactionCreationError,
+    TransactionCreationRequestSerializer,
+    TransactionSerializer
+)
+from enterprise_subsidy.apps.subsidy.api import get_subsidy_by_uuid
+from enterprise_subsidy.apps.subsidy.constants import (
+    PERMISSION_CAN_CREATE_TRANSACTIONS,
+    PERMISSION_CAN_READ_ADMIN_TRANSACTIONS,
+    PERMISSION_CAN_READ_TRANSACTIONS
+)
+from enterprise_subsidy.apps.subsidy.models import ContentNotFoundForCustomerException, Subsidy
+
+logger = logging.getLogger(__name__)
+
+
+class TransactionBaseViewMixin:
+    """
+    Base view mixin that defines default authentication and permission classes;
+    a subsidy-scoped Transaction queryset; and a default Transaction serializer.
+    """
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = TransactionSerializer
+    queryset = Transaction.objects.all()
+
+    @property
+    def subsidy_uuid(self):
+        """
+        Returns the requested ``subsidy_uuid`` path parameter.
+        """
+        return self.kwargs.get('subsidy_uuid')
+
+    @cached_property
+    def subsidy(self):
+        """
+        Returns the Subsidy instance from the requested ``subsidy_uuid``.
+        """
```

Comment:
Nit: add documentation that this raises Subsidy.DoesNotExist (which bubbles up as a 404 response?)

Reply:
the auth is actually going to fail first and result in a 403, I think all the code is now updated to reflect that.